### PR TITLE
Add remark field

### LINKED ISSUE #97: 📖 As new user, I can add remarks to a person
...so that I can take note of their preferences (dietary, allergy, etc).

________________________________________________________________
### Additional context
For now, will be adding remarks as a long String.

### PR DESCRIPTION
**Pull request by Ellevy**
Added optional Remark field for contacts as a String.
Constraints used are same as Address field.

Yet to update UI to display remark as command output. Remark currently displayed in the person card (along with the address, birthday, phone number, email, tags). 

Fixes #97

### CODE REVIEW

File: src/main/java/seedu/partyplanet/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -88,6 +89,24 @@ public static Address parseAddress(String address) throws ParseException {
         return new Address(trimmedAddress);
     }
 
+    /**
+     * Parses a {@code String remark} into a {@code remark}.
```
Comment (pyuxiang) [inline src/main/java/seedu/partyplanet/logic/parser/ParserUtil.java:93]:
```suggestion
     * Parses a {@code String remark} into a {@code Remark}.
```

File: src/main/java/seedu/partyplanet/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -88,6 +89,24 @@ public static Address parseAddress(String address) throws ParseException {
...
+     * Returns default empty remark if remark is not specified.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code remark} is invalid.
```
Comment (pyuxiang) [inline src/main/java/seedu/partyplanet/logic/parser/ParserUtil.java:97]:
```suggestion
     * @throws ParseException if the given {@code Remark} is invalid.
```

**Review by pyuxiang — CHANGES_REQUESTED** [review]
Other than the minor typos, LGTM!

**Review by pyuxiang — DISMISSED** [review]
Nice

**Review by zhengruoxin — CHANGES_REQUESTED** [review]
Just a missing space, LGTM!

File: src/main/java/seedu/partyplanet/model/person/Remark.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import static java.util.Objects.requireNonNull;
+import static seedu.partyplanet.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's remark in PartyPlanet.
+ * Guarantees: immutable; is valid as declared in {@link #isValidRemark(String)}
+ */
+public class Remark {
+
+    public static final String MESSAGE_CONSTRAINTS = "Remark can take any values, and it should not be blank";
```
[the remark this answers is not in the document]
Comment (pyuxiang) [reply src/main/java/seedu/partyplanet/model/person/Remark.java:12]: Perhaps best of both worlds: "You can write anything in the remarks, as long as it is not blank".

[the remark this answers is not in the document]
Comment (garyljj) [reply src/main/java/seedu/partyplanet/model/person/Remark.java:12]: I think she was following the wording of constraint for address field. Should we change that too? or leave both

**Review by garyljj — APPROVED** [review]
LGTM

**Review by zhengruoxin — APPROVED** [review]

**Pull request merged**